### ISSUE TITLE
Trivy is not generating compliant SPDX JSON

### ISSUE THREAD

**Issue by beltran-rubo** [aquasecurity/trivy#3267]
## Description

The generated SPDX JSON file is not compliant with SPDX format. There are two issues why the output file does not pass the [SPDX validator tool](https://tools.spdx.org/app/validate/):
- The `licenseConcluded` and `licenseDeclared` is a comma separated list of licenses instead of supporting the [official operators](https://github.com/spdx/Spdx-Java-Library/blob/cd1b91df7087c7167edafbf3413ead8ff88f552c/src/main/java/org/spdx/library/model/license/LicenseExpressionParser.java#L57).
- There are licenses that are not part of the [SPDX identifiers](https://spdx.org/licenses/). E.g. `GPL-3+ with Autoconf exception` that it should be `GPL-3.0-with-autoconf-exception`.

## What did you expect to happen?

The output of a SBOM in SPDX JSON format should be compliant with the format and pass the [SPDX validator tool](https://tools.spdx.org/app/validate/). The same also happen with SPDX key-value format.

## What happened instead?

The SPDX Json format is not compliant with SPDX. Other tools like `syft` report the same licenses but in a compliant format, using `AND` for concatenating the licenses.

## Output of `trivy -v`:

```
Version: 0.35.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-12-05 06:07:28.002545547 +0000 UTC
  NextUpdate: 2022-12-05 12:07:28.002545147 +0000 UTC
  DownloadedAt: 2022-12-05 10:38:15.116911974 +0000 UTC
```

## Additional details (base image name, container registry info...):

Tested with Dockerhub Debian image E.g. `./trivy -q i --format spdx-json debian:latest`

**Comment by masahiro331** [comment]
I will check it.

**Comment by tokcum** [comment]
Hey, I see so many helpful features being delivered with any new trivy release. However, this PR for SPDX compliant license expressions is still not merged.

I'm contributing to CycloneDX's rust-cargo project and invalid SPDX license expressions are an issue because de-serialization fails. We've already discussed if we could be less strict when handling license expressions. However, we concluded that this wouldn't help interoperability and even users rely on well-defined license identifiers.

It would be great to see this PR merged soon. Thanks.
